miden-proving-service-client/src/prover.rs

```diff
@@ -80,8 +80,9 @@ impl TransactionProver for RemoteTransactionProver {
         let mut client = self
             .client
             .write()
-            .take()
-            .ok_or_else(|| TransactionProverError::other("client should be connected"))?;
+            .as_ref()
+            .ok_or_else(|| TransactionProverError::other("client should be connected"))?
+            .clone();
```

Comment:
Cloning the client is pretty light-weight, right?

Reply:
Yes, it is composed only for the tonic client which isn't "big".